staging/src/github.com/kubeedge/beehive/pkg/core/context/context_channel.go

```diff
@@ -150,8 +150,18 @@ func (ctx *ChannelContext) SendResp(message model.Message) {
 
 	ctx.anonChsLock.RLock()
 	defer ctx.anonChsLock.RUnlock()
+
+	deadline := time.Now().Add(MessageTimeoutDefault)
+	sendTimer := time.NewTimer(time.Until(deadline))
+
 	if channel, exist := ctx.anonChannels[anonName]; exist {
-		channel <- message
+		select {
+		case channel <- message:
+		case <-sendTimer.C:
+			klog.Warning("timeout when SendResp message")
+			return
+		}
+		sendTimer.Stop()
```

Comment:
Thanks for the fixing, small question, Here the message just be written to the channel, will it be blocked?

Reply:
Channel operations are blocking in nature, when some data is written to the channel, goroutine is blocked until some other goroutine reads it from that channel, so if no one reads it, it will be blocked.

Reply:
SendSync already has a timer. Do we need another timer here?